homeassistant/components/nad/media_player.py

```diff
@@ -64,65 +64,42 @@
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the NAD platform."""
-    if config.get(CONF_TYPE) == "RS232":
+    if config.get(CONF_TYPE) in ("RS232", "Telnet"):
         add_entities(
-            [
-                NAD(
-                    config.get(CONF_NAME),
-                    NADReceiver(config.get(CONF_SERIAL_PORT)),
-                    config.get(CONF_MIN_VOLUME),
-                    config.get(CONF_MAX_VOLUME),
-                    config.get(CONF_SOURCE_DICT),
-                )
-            ],
-            True,
-        )
-    elif config.get(CONF_TYPE) == "Telnet":
-        add_entities(
-            [
-                NAD(
-                    config.get(CONF_NAME),
-                    NADReceiverTelnet(config.get(CONF_HOST), config.get(CONF_PORT)),
-                    config.get(CONF_MIN_VOLUME),
-                    config.get(CONF_MAX_VOLUME),
-                    config.get(CONF_SOURCE_DICT),
-                )
-            ],
-            True,
+            [NAD(config)], True,
         )
     else:
         add_entities(
-            [
-                NADtcp(
-                    config.get(CONF_NAME),
-                    NADReceiverTCP(config.get(CONF_HOST)),
-                    config.get(CONF_MIN_VOLUME),
-                    config.get(CONF_MAX_VOLUME),
-                    config.get(CONF_VOLUME_STEP),
-                )
-            ],
-            True,
+            [NADtcp(config)], True,
         )
 
 
 class NAD(MediaPlayerEntity):
     """Representation of a NAD Receiver."""
 
-    def __init__(self, name, nad_receiver, min_volume, max_volume, source_dict):
+    def __init__(self, config):
         """Initialize the NAD Receiver device."""
-        self._name = name
-        self._nad_receiver = nad_receiver
-        self._min_volume = min_volume
-        self._max_volume = max_volume
-        self._source_dict = source_dict
+        self.config = config
+        self._instantiate_nad_receiver()
+        self._min_volume = config.get(CONF_MIN_VOLUME)
+        self._max_volume = config.get(CONF_MAX_VOLUME)
+        self._source_dict = config.get(CONF_SOURCE_DICT)
         self._reverse_mapping = {value: key for key, value in self._source_dict.items()}
 
         self._volume = self._state = self._mute = self._source = None
 
+    def _instantiate_nad_receiver(self) -> NADReceiver:
+        if self.config.get(CONF_TYPE) == "RS232":
```

Comment:
Please use `dict[key]` for required config schema keys and keys with default schema values.

```suggestion
        if self.config[CONF_TYPE] == "RS232":
```